python/lsst/afw/cameraGeom/basicCameraFactory.py

```diff
@@ -0,0 +1,225 @@
+#
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+from lsst.afw.cameraGeom import makeCameraFromCatalogs
+from lsst.afw.cameraGeom import (DetectorConfig, CameraConfig, PUPIL, FOCAL_PLANE, PIXELS)
+
+
+__all__ = ["BasicCameraFactory"]
+
+
+class BasicCameraFactory(object):
+    """
+    This class reads in a text file describing the layout of a camera's focal plane
+    and creates and afwCameraGeom.camera instantiation that contains all of the
+    coordinate transformations (i.e. between PUPIL, FOCAL_PLANE, PIXELS, and TAN_PIXELS
+    coordinates) in that focal plane.  The resulting camera will not contain any
+    amplifier information.  Cameras produced by this method should only be used for
+    mapping objects' positions on the sky to positions on the focal plane.
+
+    To create a camera, one should:
+
+    cameraFactory = BasicCameraFactory('path/to/focal/plane/layout.txt')
+    camera = cameraFactory.makeCamera()
+
+    The input text file should consist of one row for each detector.  Each row should contain
+
+    Name of the detector
+    The x coordinate of the detector's center in microns (relative to the center of the focal plane)
+    The y coordinate of the detector's center in microns
+    The size of each pixel in microns
+    The number of x pixels
+    The number of y pixels
+    A string indicating what type of detector it is (SCIENCE, FOCUS, GUIDER, or WAVEFRONT)
+    Three euler angles (in degrees) governing the orientation of the detector
+
+    For example:
+
+    #
+    # rows are the chips in the focal plane
+    # columns are the name, x position (microns), y position (microns),
+    # pixel size (microns), number of x pixels, number of y pixels,
+    # group of sensors, 3 euler rotations (degrees)
+    #
+    Det00 0.0 0.0 2.0 400 400 science 20.0 0.0 0.0
+    Det01 0.0 9000.0 2.0 400 400 science 10.0 0.0 0.0
+    Det02 -9000.0 0.0 2.0 400 400 wave 30.0 0.0 0.0
+    """
+
+    def __init__(self, detectorLayoutFile,
+                 detectorIdFromAbbrevName=None,
+                 detTypeMap=None,
+                 radialTransform=[0.0, 1.0, 0.0, 0.0],
+                 cameraName='LSST'
+                 ):
+
+        """
+        @param [in] detectorLayoutFile is the absolute path to the file
+        listing the layout of all of the chips in the camera's focal plane
+        (see this class' docstring for detailed information on the contents
+        of that file).
+
+        @param [in] detectorIdFromAbbrevName is a (required) method mapping the
+        names of detectors as stored in the detectorLayoutFile to unique integers
+        identifying each detector.
+
+        @param[in] detTypeMap is a dict mapping the group name of the sensors in
+        detectorLayoutFile to the integers:
+
+        0 = SCIENCE
+        1 = FOCUS
+        2 = GUIDER
+        3 = WAVEFRONT
+
+        if left as None, this will just try to cast the contents of detectorLayoutFile
+        as an int
```

Comment:
The other obvious option here is to assume all sensors are SCIENCE.  I'm not sure which is better.